test/events.js

```diff
@@ -0,0 +1,95 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const assert = require('insist');
+const crypto = require('crypto');
+const fs = require('fs');
+const path = require('path');
+
+const db = require('../lib/db');
+const events = require('../lib/events');
+const P = require('../lib/promise');
+
+const UID = crypto.randomBytes(16).toString('hex');
+const mock = require('./lib/mock')({ userid: UID });
+const Server = require('./lib/server');
+const Static = require('./lib/static');
+
+const imagePath = path.join(__dirname, 'lib', 'firefox.png');
+const imageData = fs.readFileSync(imagePath);
+
+const SIZES = require('../lib/img').SIZES;
+const SIZE_SUFFIXES = Object.keys(SIZES).map(function(val) {
+  if (val === 'default') {
+    return '';
+  }
+  return '_' + val;
+});
+
+
+/*global describe,it,before*/
+
+describe('events', function() {
+  describe.only('onDeleteMessage', function() {
+    var tok = crypto.randomBytes(32).toString('hex');
+    function Message(del) {
+      return {
+        event: 'delete',
+        uid: UID + '@accounts.firefox.com',
+        del: del
+      };
+    }
+
+    before(function() {
+      this.slow(2000);
+      this.timeout(3000);
+      mock.token({
+        user: UID,
+        email: 'user@example.domain',
+        scope: ['profile:avatar:write']
+      });
+      mock.image();
+      return Server.api.post({
+        url: '/avatar/upload',
+        payload: imageData,
+        headers: { authorization: 'Bearer ' + tok,
+          'content-type': 'image/png',
+          'content-length': imageData.length
+        }
+      }).then(function(res) {
+        assert.equal(res.statusCode, 201);
+        assert(res.result.url);
+        assert(res.result.id);
+        return res.result.url;
+      }).then(function(s3url) {
+        return P.all(SIZE_SUFFIXES).map(function(suffix) {
+          return Static.get(s3url + suffix);
+        });
+      }).then(function(responses) {
+        assert.equal(responses.length, SIZE_SUFFIXES.length);
+        responses.forEach(function(res) {
+          assert.equal(res.statusCode, 200);
+        });
+      });
+    });
+
+    it('should delete avatars', function(done) {
+      mock.deleteImage();
+      events.onData(new Message(function() {
+        db.getAvatars(UID).then(function(avatars) {
+          assert.equal(avatars.length, 0);
+        }).done(done, done);
+      }));
+    });
+    it('should delete display name', function(done) {
+      db.setDisplayName(UID, 'foo bar').then(function() {
+        events.onData(new Message(function() {
+          db.getDisplayName(UID).then(function(profile) {
+            assert.equal(profile.displayName, undefined);
+          }).done(done, done);
+        }));
+      });
+    });
+  });
+});
```

Comment:
Please add some edge-case-handling tests, in particular:
* non-"delete" messages are discarded
* errors during deletion result in the message staying in the queue for eventual retry